js/src/logger.ts

```diff
@@ -855,9 +875,8 @@ class BackgroundLogger {
           break;
         }
 
-        const itemS = JSON.stringify(item);
-        items.push(itemS);
-        itemsLen += itemS.length;
+        items.push(item);
```

Comment:
Hmm so now we're stringifying the JSON objects twice: once for computing `itemsLen` and then another time when creating `dataS`. It might be preferable to fake the construction of `dataS` as
```
dataS = `{"rows": ${constructJsonArray(itemsS)}, "api_version": 2}`
```

Reply:
I recognize that we have to manipulate the object itself for the legacy endpoint, but maybe we can only incur the extra work there. Like we could keep track of both `items` and `itemsS`, and fallback to patching and re-stringifying `items` if they are on the old endpoint.